compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -165,6 +165,8 @@ generateLoad32BitConstant(TR::CodeGenerator* cg, TR::Node* node, int32_t value,
             return generateRegLitRefInstruction(cg, TR::InstOpCode::L, node, targetRegister, value, TR_DataAddress, dependencies, cursor, literalPoolRegister);
          if (sym->isCountForRecompile())
             return generateRegLitRefInstruction(cg, TR::InstOpCode::L, node, targetRegister, value, TR_GlobalValue, dependencies, cursor, literalPoolRegister);
+         if (sym && sym->isRecompilationCounter())
+            return generateRegLitRefInstruction(cg, TR::InstOpCode::L, node, targetRegister, value, TR_BodyInfoAddress, dependencies, cursor, literalPoolRegister);
```

Comment:
So is this all that was needed to support loading statics directly for AOT on Z?

Reply:
I think not adding this relocation here was always a bug. But we never encountered it because we always chose the indirect trees in AOT compilation. 

For context this ended up being a pretty elusive issue to narrow down. Mainly because when trying to narrow down to a method (or narrow down the problem using any of the usual JIT options), the issue would disappear.  But yeah this is all that's needed. I haven't seen any other failures yet. Will report if I see anything else by EOD today.